u_parrots.go

```diff
@@ -250,15 +252,17 @@ func (uconn *UConn) ApplyPreset(p *ClientHelloSpec) error {
 		uconn.greaseSeed[ssl_grease_extension2] ^= 0x1010
 	}
 
-	hello.CipherSuites = p.CipherSuites
+	hello.CipherSuites = make([]uint16, len(p.CipherSuites))
```

Comment:
This is confusing.  Are you deep-copying the input (in which case it's safe to use the same input for many calls) or are you requiring the input to be independent for each call (in which case there's no need to make a copy).

Reply:
The only reliable way to deepcopy all the extensions I found is to add `Clone()` method to `TLSExtension` interface. Which I didn't quite want to do.
Current solution simply makes a new preset for built-in parrots. For custom ones, only shared state is pointers and slices in fields of `TLSExtension`s, which I indeed could be more explicit about.